src/composables/addressBook.ts

```diff
@@ -0,0 +1,234 @@
+import { computed, ref } from 'vue';
+import { Directory, Filesystem } from '@capacitor/filesystem';
+import { Capacitor } from '@capacitor/core';
+
+import {
+  AccountAddress,
+  IAddressBook,
+  IAddressBookEntry,
+  Protocol,
+} from '@/types';
+import { STORAGE_KEYS, MODAL_ADDRESS_BOOK_IMPORT } from '@/constants';
+import { AddressBookEntryExists, AddressBookInvalidAddress, AddressBookRequiredFields } from '@/lib/errors';
+import {
+  convertBlobToBase64,
+  getProtocolByAddress,
+  handleUnknownError,
+  pipe,
+} from '@/utils';
+import { tg as t } from '@/popup/plugins/i18n';
+
+import { createCustomScopedComposable } from './composablesHelpers';
+import { useStorageRef } from './storageRef';
+import { useModals } from './modals';
+import { useFileDialog } from './fileDialog';
+
+interface IAddressBookOptions {
+  name: string;
+  address: AccountAddress;
+  isBookmarked?: boolean;
+}
+
+const addressBook = useStorageRef<IAddressBook>([], STORAGE_KEYS.addressBook);
+
+export const useAddressBook = createCustomScopedComposable(() => {
+  const { openModal, openDefaultModal } = useModals();
+  const { openFileDialog } = useFileDialog();
+
+  const activeFilter = ref<Protocol | null>(null);
+  const searchQuery = ref<string>('');
+  const showBookmarked = ref(false);
+
+  function filterAddressBookByBookmarked(entries: IAddressBookEntry[]) {
+    return showBookmarked.value ? entries.filter((entry) => entry.isBookmarked) : entries;
+  }
+  function filterAddressBook(entries: IAddressBookEntry[]) {
+    return activeFilter.value && !showBookmarked.value
+      ? entries.filter((entry) => entry.protocol === activeFilter.value)
+      : entries;
+  }
+  function filterAddressBookBySearchPhrase(entries: IAddressBookEntry[]) {
+    const searchQueryLower = searchQuery.value.toLowerCase();
+    return entries.filter((entry) => {
+      const nameLower = entry.name.toLowerCase();
+      const addressLower = entry.address.toLowerCase();
+      return nameLower.includes(searchQueryLower) || addressLower.includes(searchQueryLower);
+    });
```

Comment:
What do you think about this approach:

```suggestion
    return entries.filter(({ name, address }) => [name, address].some((val) => val.toLowerCase().includes(searchQueryLower);
```